packages/atlas/src/components/_video/VideoPlayer/VideoOverlay.tsx

```diff
@@ -31,28 +34,41 @@ export const VideoOverlay: FC<VideoOverlayProps> = ({
   isPlayNextDisabled,
   isMinimized,
   playRandomVideoOnEnded = true,
+  currentVideoCreatedAt,
 }) => {
   const [randomNextVideo, setRandomNextVideo] = useState<BasicVideoFieldsFragment | null>(null)
-  const { videos } = useBasicVideos(
+  const { videos, refetch } = useBasicVideos(
     {
+      limit: 1,
+      orderBy: VideoOrderByInput.ChannelCreatedAtAsc,
       where: {
+        ...publicVideoFilter,
         channel: {
           id_eq: channelId,
         },
+        createdAt_gt: currentVideoCreatedAt,
       },
     },
-    { context: { delay: 2000 } }
+    { notifyOnNetworkStatusChange: true }
   )
 
   useEffect(() => {
-    if (!videos?.length || videos.length <= 1) {
+    if (!videos?.length || videos.length === 0) {
+      refetch({
+        where: {
+          channel: {
+            id_eq: channelId,
+          },
+          createdAt_lt: currentVideoCreatedAt,
+        },
+      })
```

Comment:
Sorry, I missed that before. I tested this on video /video/552 and I think this is not working correctly. Channel has two videos and it's not switching to the other one.